c++/src/kj/async.h

```diff
@@ -27,6 +27,23 @@
 
 KJ_BEGIN_HEADER
 
+#ifndef KJ_USE_FIBERS
+#if __BIONIC__ || __FreeBSD__ || __OpenBSD__
+// These platforms don't support fibers.
+#define KJ_USE_FIBERS 0
+#else
+#ifndef KJ_NO_EXCEPTIONS
+#define KJ_USE_FIBERS 1
+#else
+#define KJ_USE_FIBERS (!KJ_NO_EXCEPTIONS)
+#endif /* KJ_NO_EXCEPTIONS */
+#endif /* unsupported platforms */
+#else
+#if defined(KJ_NO_EXCEPTIONS) && KJ_NO_EXCEPTIONS && KJ_USE_FIBERS
+#error Fibers are not implemented because exceptions are disabled
+#endif /* KJ_NO_EXCEPTIONS */
+#endif /* KJ_USE_FIBERS */
```

Comment:
Let's use indentation here, and simplify things a bit. There's no need to check if KJ_NO_EXCEPTIONS is defined; if it's not defined, the preprocessor treats it as falsy. We generally use `#if KJ_NO_EXCEPTIONS` without first checking if it's defined elsewhere in KJ code.

```suggestion
#ifndef KJ_USE_FIBERS
  #if __BIONIC__ || __FreeBSD__ || __OpenBSD__ || KJ_NO_EXCEPTIONS
    // These platforms don't support fibers.
    #define KJ_USE_FIBERS 0
  #else
    #define KJ_USE_FIBERS 1
  #endif
#else
  #if KJ_NO_EXCEPTIONS && KJ_USE_FIBERS
    #error "Fibers cannot be enabled when exceptions are disabled."
  #endif
#endif
```